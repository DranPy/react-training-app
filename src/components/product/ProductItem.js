import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import ProductActions from './ProductActions';

class ProductItem extends Component {
  static propTypes = {
    product: PropTypes.object,
  };

  render() {
    const { id, name, price, imgUrl } = this.props.product;
    const currency = this.props.currency || '$';

    return (
      <li className="col-3">
        <div className="product-item">
          <Link to={`/products/${id}`}>
            <img src={imgUrl} className="product-item__img img-fluid rounded" alt={name} />
          </Link>
          <div>
            <h5 className="product-item__name">
              <Link to={`/products/${id}`}>{name}</Link>
            </h5>
            <div className="product-item__price">
              <strong>{price + currency}</strong>
            </div>
          </div>
          <ProductActions />
        </div>
      </li>
    );
  }
}

export default ProductItem;
