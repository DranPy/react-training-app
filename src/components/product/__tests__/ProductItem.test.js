import { shallow } from 'enzyme';
import React from 'react';

import ProductItem from '../ProductItem';

describe('<ProductItem />', () => {
  const defaultProps = {
    product: {
      id: '1',
      name: 'test name 1',
      category: 'category 1',
      price: '1$',
      imgUrl: 'src/1.img',
    },
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<ProductItem {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
