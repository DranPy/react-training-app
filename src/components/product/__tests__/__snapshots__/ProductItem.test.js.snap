// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductItem /> should match snapshot 1`] = `
<li
  className="product-item"
>
  <Link
    replace={false}
    to="/products/1"
  >
    <img
      alt="test name 1"
      className="product-item__img img-fluid rounded"
      src="src/1.img"
    />
  </Link>
  <div>
    <h5
      className="product-item__name"
    >
      <Link
        replace={false}
        to="/products/1"
      >
        test name 1
      </Link>
    </h5>
    <div
      className="product-item__price"
    >
      1$$
    </div>
  </div>
  <div
    className="product-item__actions btn-group"
  >
    <button
      className="btn btn-primary"
      title="Buy"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="shopping-cart"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <button
      className="btn btn-outline-primary"
      title="Remove"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="times-circle"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
    <button
      className="btn btn-outline-primary"
      title="Edit"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="pen"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
  </div>
</li>
`;
